components/Cards/src/MDCCard.m

```diff
@@ -221,16 +221,12 @@ - (UIColor *)shadowColorForState:(UIControlState)state {
 }
 
 - (void)setHighlighted:(BOOL)highlighted {
-  [super setHighlighted: highlighted];
-  if (highlighted) {
-    // Note: setHighlighted: can get getting more calls with YES than NO when clicking rapidly.
-    // To guard against ink never going away and darkening our card we call
-    // startTouchEndedAnimationAtPoint:completion:.
-    [self.inkView startTouchEndedAnimationAtPoint:_lastTouch completion:nil];
+  if (highlighted && !self.highlighted) {
```

Comment:
Can this be consolidated to a single
if (highlighted != self.highlighted) {
  [self.inkView startTouchBegan...];
}

Reply:
there are two different options when this happens. so we could either leave as is where `if (highlighted && !self.highlighted) { } else if (!highlighted && self.highlighted) { }` or alternatively `if (highlighted != self.highlighted) { if (highlighted) { } else { } }`. Although the latter would cause another branching potentially

Reply:
I'm not sure I follow, it looks like we have two clauses in an if statement that are executing the same block of code.

I would expect these two clauses to be combined into a single if statement.
```
if ((highlighted && !self.highlighted) || 
    (!highlighted && self.highlighted) {
  [self.inkView startTouchEndedAnimationAtPoint:_lastTouch completion:nil];
}
```

Alternatively, as both highlighted and self.highlighted are BOOLs, if there are no side effects from the getter we can compare them directly.
```
if (highlighted != self.highlighted) {
  [self.inkView startTouchEndedAnimationAtPoint:_lastTouch completion:nil];
}
```

Reply:
They are both different methods in the conditional clauses, one is `startTouchBegan...` the other is `startTouchEnded...`